### ISSUE TITLE
Result should require a non empty value

### ISSUE THREAD

**Issue by carlesba** [rametta/pratica#29]
I think Result should require a defined value on its creation in order to guarantee a value in `cata`. 

ie, 
```ts

const format= (text: string): Result<number, string> => {
  const number = Number(text)
  return isNaN(number)? Err('bad input') : Ok(number)
}

format('2,000').cata({
  Ok: (result)=> // result is not guaranteed, can be undefined
  Err:// ...
})
```

format's output is a Result number, but still, we have to add a conditional guard to fix the type.

**Comment by carlesba** [comment]
hey Jason, thank you for taking the time to review this. 
How would you use NonNullable in this case?

**Comment by csfalcione** [comment]
I also ran into this issue. What do you think of https://github.com/rametta/pratica/pull/31?

**Comment by rametta** [comment]
> I also ran into this issue. What do you think of #31?

Looks good, just a few comments on it then we can merge

**Comment by rametta** [comment]
This is now fixed and available in version 2.1.0 on npm.
